### PR TITLE
docs: few-shot conceptual guide

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

**Review by ccurme — APPROVED** [review]

File: docs/docs/concepts.mdx [hunk 1]
```diff
@@ -1101,6 +1104,83 @@ The following how-to guides are good practical resources for using function/tool
...
 For a full list of model providers that support tool calling, [see this table](/docs/integrations/chat/#advanced-features).
 
+### Few-shot prompting
+
+One of the most effective ways to improve model performance is to give a model examples of what you want it to do. The technique of adding example inputs and expected outputs to a model prompt is known as "few-shot prompting". There are a few things to think about when doing few-shot prompting:
+
+1. How are examples generated,
+2. How many examples are in each prompt,
+3. How are examples selected at runtime,
+4. How are examples formatted in the prompt.
```
Comment (jakerachleff) [inline docs/docs/concepts.mdx:1114]:
```suggestion
1. How are examples generated?
2. How many examples are in each prompt?
3. How are examples selected at runtime?
4. How are examples formatted in the prompt?
```

File: docs/docs/concepts.mdx [hunk 2]
```diff
@@ -1101,6 +1104,83 @@ The following how-to guides are good practical resources for using function/tool
...
+
+Here are the considerations for each.
+
+#### 1. Generating examples
+
+The first and most important step of few-shot prompting is coming up with a good dataset of examples.
+The key to a good dataset is that the examples are relevant, clear and informative.
+The example inputs should be similar to the types of inputs you'll see at runtime, it should be clear what the correct answer was and why, and they should be examples that aren't obvious.
```
Comment (jakerachleff) [inline docs/docs/concepts.mdx:1122]:
```
```suggestion
The first and most important step of few-shot prompting is coming up with a good dataset of examples. Good examples should be relevant at runtime, clear, informative, and provide information that was not already known to the model.
```

File: docs/docs/concepts.mdx [hunk 3]
```diff
@@ -1101,6 +1104,83 @@ The following how-to guides are good practical resources for using function/tool
...
+
+At a high-level, the basic ways to generate examples are:
+- Manual: a person/people generates examples they think are useful,
```
Comment (jakerachleff) [inline docs/docs/concepts.mdx:1125]: overall nit (won't correct every instance) but bullets shouldn't end in ,

File: docs/docs/concepts.mdx [hunk 4]
```diff
@@ -1101,6 +1101,83 @@ The following how-to guides are good practical resources for using function/tool
...
+- Better model: a better (presumably more expensive/slower) model's responses are used as examples for a worse (presumably cheaper/faster) model,
+- User feedback: users leave feedback on interactions with the application and positive feedback is automatically added to the dataset of examples,
```
[the remark this answers is not in the document]
Comment (jakerachleff) [reply docs/docs/concepts.mdx:1127]: speaking of which, any mention on like when to use positive/negative examples vs just positive?

File: docs/docs/concepts.mdx [hunk 5]
```diff
@@ -1101,6 +1104,83 @@ The following how-to guides are good practical resources for using function/tool
 
 For a full list of model providers that support tool calling, [see this table](/docs/integrations/chat/#advanced-features).
 
+### Few-shot prompting
```
Comment (jakerachleff) [inline docs/docs/concepts.mdx:1107]: Should there be some information on _why_ few shot prompting is important?

Comment (baskaryan) [reply docs/docs/concepts.mdx:1107]: as in beyond the first sentence?

**Pull request merged**